### PR TITLE
Run all scripts and build dist on Travis

### PR DESCRIPTION
**Pull request by RubenVerborgh**
This pull request introduces two major changes:
- npm scripts are reorganized and renamed for consistency
- the `dist` folder is removed (and ignored) from `master`, and instead built automatically by Travis in another branch

On every push to `master`, an automated commit to the other branch is performed, labeled `Update to {commit_hash}.` This avoids the need of the heavy `dist` folder in the development code, reducing weight and merge conflicts.

Open question: to what branch should the automatic build be pushed?
1. to `gh_pages`: in this case, I propose clearing the existing `gh_pages` branch. Benefit: the resulting script is accessible from http://linkeddata.github.io/rdflib.js/dist/rdflib.min.js
2. to another branch: in this case, could you suggest a name?

### CODE REVIEW

File: .travis_publish [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Update pages
+rm -r * 2> /dev/null
+cp -r $REPO_PATH/dist .
+
+# Commit and push latest version
+git add .
+git config user.name  "Travis"
+git config user.email "travis@travis-ci.org"
+git commit -m "Update to $TRAVIS_COMMIT."
+git push -fq origin $PUBLICATION_BRANC 2>&1 > /dev/null
```
Comment (dmitrizagidulin) [inline .travis_publish]: I suspect that's missing an 'H'?

**Comment by dmitrizagidulin** [comment]
Does this build dist on every commit to master, CI style? Or does it only build on npm version changes?

File: .travis_publish (hunk 1 above)
Comment (RubenVerborgh) [reply .travis_publish]: Aw, yes. Good catch. Trouble is I can't test this thing from a branch.

**Comment by RubenVerborgh** [comment]
Yes, it will build on every commit to master. I can make it only build on tags if you want.

**Comment by dmitrizagidulin** [comment]
I don't know, what do you think is best? On the one hand, seems like every commit is overkill. On the other, hey, Travis has the "CI" part right in the name :)

**Comment by RubenVerborgh** [comment]
In my projects, I build on every commit to master. However, I tend to do development in a separate branch, so in practice, master is only updated on every npm version.

**Comment by dmitrizagidulin** [comment]
Seems legit.

**Comment by timbl** [comment]
I agree pushing every master commit to cleared out gh-pages

**Comment by RubenVerborgh** [comment]
Okay, I will prepare `gh-pages` shortly, double-check, and then merge this PR. (Might still require some tweaks, but I can only see the effect once the CI starts running.)

**Comment by RubenVerborgh** [comment]
Worked on first pass 😄  Here's the first automated commit: 63feeaa20e6bb9e3f3f3de4142778f56854e8231

@dmitrizagidulin Could you please set the `github.io` branch to `gh_pages`  (it's currently `master`, I believe).

**Comment by dmitrizagidulin** [comment]
@RubenVerborgh it's now set to `gh_pages`.

**Comment by RubenVerborgh** [comment]
@dmitrizagidulin Thanks!

**Pull request merged**